### PR TITLE
For #35896, more debugging and fixes

### PR DESCRIPTION
**Pull request by jfboismenu**
- Added some debugging messages
- Workspaces without a host can be used from any machine, so if a user owns some of these, they will be shown in the workspace selection dialog.
- Fixed a few crashing errors
- Documented all methods.

Note: See second commit to avoid trailing whitespace changes. The very vast majority of these were on lines with comments or empty lines and almost no lines with actual code had them, so I simply stripped all of them.

### CODE REVIEW

File: python/connection/connection.py [hunk 1]
```diff
@@ -19,40 +19,44 @@
...
 from P4 import P4, P4Exception
 
 from .user_settings import UserSettings
 
+
 class SgtkP4Error(TankError):
     """
     Specialisation of TankError raised after catching and processing a P4Exception
     """
-    pass
```
Comment (thebeeland) [inline python/connection/connection.py]: This doesn't require a pass to work? Is a docstring enough to not cause a whitespace exception?

File: python/connection/connection.py [hunk 2]
```diff
@@ -585,13 +611,17 @@ def _do_connect_and_login(self, widget):
...
                                           ("Failed to log-in user '%s' to the Perforce server:\n\n    '%s'\n\n%s"
-                                          % (widget.user, server, e)))
+                                           % (widget.user, server, e)))
             return False
 
         return True
 
     def _get_current_workspace(self):
         """
+        Returns the current workspace based on the framework's settings or P4CLIENT is no setting
```
Comment (thebeeland) [inline python/connection/connection.py]: "...if no setting..."

File: python/connection/connection.py [hunk 3]
```diff
@@ -527,6 +547,9 @@ def _on_browse_workspace(self, widget):
 
     def _on_open_connection(self, widget):
         """
+        Called when the user clicks Connected on the connection dialog.
+
+        :param widget: Dialog object.
```
Comment (thebeeland) [inline python/connection/connection.py]: Probably QDialog, right?

File: python/connection/connection.py (hunk 1 above)
Comment (jfboismenu) [inline python/connection/connection.py]: Yup.

**Pull request closed without merging**